sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/DotNetSensor.java

```diff
@@ -76,11 +76,12 @@ public void describe(SensorDescriptor descriptor) {
   public void execute(SensorContext context) {
     FileSystem fs = context.fileSystem();
     boolean hasMainFiles = SensorContextUtils.hasFilesOfType(fs, Type.MAIN, pluginMetadata.languageKey());
+    boolean hasTestFiles = SensorContextUtils.hasFilesOfType(fs, Type.TEST, pluginMetadata.languageKey());
     boolean hasProjects = projectTypeCollector.hasProjects();
-    if (hasMainFiles && hasProjects) {
+    if ((hasMainFiles || hasTestFiles) && hasProjects) {
```

Comment:
Is this possible, to "have projects" and not have code at all? I guess that would be possible with an empty solution... can you add an integration test for this case? Or maybe we have one already?

In any case, I think this is really an edge case (which should be logged inside "importResults", if there are no results (or the results are empty).

It would be simpler to check `boolean hasFiles = SensorContextUtils.hasFilesOfType(fs, pluginMetadata.languageKey());`  and then `if (hasFiles && hasProjects)`.

Maybe it will be easier to talk directly and decide the logic here should be like. Because:
- we'd like to log if in the whole solution there's only TEST files (probably a config error)
- we'd like to log a debug if there are no C#/VB files (probably not scanner for .NET, or a solution with empty projects, or just TypeScript code for example?)

Reply:
In the end, we may want to keep `(hasMainFiles || hasTestFiles)`, but let's think how to make the code more readable.

Reply:
Regarding the analysis of Empty projects. It is not possible to analyze empty projects.  "No analysable projects were found. SonarQube analysis will not be performed."